### PR TITLE
SOC-1566 | Use WhoAmI service instead of Helios for access_token validation

### PR DESCRIPTION
**Pull request by kvas-damian**
https://wikia-inc.atlassian.net/browse/SOC-1566

### CODE REVIEW

File: server/app.ts [hunk 1]
```diff
@@ -81,7 +81,7 @@ server.register(plugins, (err: any) => {
 	}
 });
 
-server.auth.scheme('helios', HeliosSession.scheme);
+server.auth.scheme('helios', WikiaSession.scheme);
```
Comment (bkoval) [inline server/app.ts]: How about switching scheme name form 'helios' to 'wikia'?

File: server/lib/WikiaSession.ts [hunk 2]
```diff
@@ -31,18 +31,18 @@ module HeliosSession {
...
+							Logger.error('WhoAmI connection error: ', {
 								err: err,
 								parseError: parseError
 							});
-							return reply(Boom.unauthorized('Helios connection error'));
+							return reply(Boom.unauthorized('WhoAmI connection error'));
 						}
 
-						if (parsed.error) {
+						if (parsed.status === 401) {
```
Comment (bkoval) [inline server/lib/WikiaSession.ts]: What if it's going to be 500 or 400? Shouldn't we return Boom.unauthorized everytime when status !== 200?

File: server/lib/WikiaSession.ts [hunk 3]
```diff
@@ -50,7 +50,13 @@ module HeliosSession {
 				}
 
 				Wreck.get(
-					authUtils.getHeliosUrl('/info') + '?code=' + accessToken + '&noblockcheck=1',
+					authUtils.getWhoAmIUrl(),
+					{
+						timeout: 3000,
```
Comment (bkoval) [inline server/lib/WikiaSession.ts]: This value could be configurable when moved to localSettings.whoAmIService :)

File: server/lib/WikiaSession.ts (hunk 2 above)
Comment (kvas-damian) [inline server/lib/WikiaSession.ts]: parsed.status != HTTP response status.
HTTP errors are handled few lines above.

**Comment by bkoval** [comment]
LGTM :santa:

File: server/lib/WikiaSession.ts (hunk 2 above)
Comment (rafalkalinski) [inline server/lib/WikiaSession.ts]: +1

File: server/lib/WikiaSession.ts [hunk 4]
```diff
@@ -50,7 +50,13 @@ module HeliosSession {
...
+						headers: {
+							Cookie: 'access_token=' + encodeURIComponent(accessToken)
```
Comment (rafalkalinski) [inline server/lib/WikiaSession.ts]: Isn't it enough to set 
xhrFields: {
	withCredentials: true,
},
?

Comment (kvas-damian) [inline server/lib/WikiaSession.ts]: It isn't frontent bro...

Comment (rafalkalinski) [inline server/lib/WikiaSession.ts]: whatever...

**Comment by rafalkalinski** [comment]
LGTM

File: config/localSettings.base.ts [hunk 5]
```diff
@@ -26,6 +26,10 @@ var localSettings: LocalSettings = {
 	discuss: {
 		baseAPIPath: 'discussion',
 	},
+	whoAmIService: {
```
Comment (rogatty) [inline config/localSettings.base.ts]: Why the `Service` prefix? We don't use it after `helios`. Would be nice to have some standard here.

Comment (kvas-damian) [inline config/localSettings.base.ts]: After looking again at ```helios``` and ```discuss``` I found that they're not enough descriptive. That's why I decided to start using ```Service``` suffix.
Changing all of them is not in scope of this ticket.

Comment (rogatty) [inline config/localSettings.base.ts]: Fair enough. Do we have a ticket to clean them up?

Comment (kvas-damian) [inline config/localSettings.base.ts]: https://wikia-inc.atlassian.net/browse/SOC-1631

**Pull request merged**